File: frontend/src/types/portfolio.ts
export interface Position {
  symbol: string;
  qty: number;
  avg_entry_price: number;
  current_price: number;
  market_value: number;
  unrealized_pl: number;
  unrealized_plpc: number;
  side: string;
}

export interface Account {
  equity: number;
  buying_power: number;
  day_pnl: number;
  portfolio_value: number;
  status: string;
}

export interface Trade {
  id: number;
  ticker: string;
  side: string;
  quantity: number;
  price: number;
  order_type: string;
  status: string;
  filled_at: string | null;
  filled_price: number | null;
  filled_quantity: number | null;
  commission: number;
  notes: string | null;
  alpaca_order_id: string | null;
  created_at: string;
}

export interface TradeHistoryItem {
  id: number;
  ticker: string;
  side: string;
  quantity: number;
  price: number;
  order_type: string;
  status: string;
  filled_at: string | null;
  filled_price: number | null;
  filled_quantity: number | null;
  commission: number;
  notes: string | null;
  created_at: string;
}

export interface TradeHistoryResponse {
  items: TradeHistoryItem[];
  total: number;
  page: number;
  limit: number;
  pages: number;
}

export interface AlpacaConfig {
  is_connected: boolean;
  paper_trading: boolean;
  last_sync: string | null;
}

export interface TradeForm {
  symbol: string;
  qty: string;
  side: "buy" | "sell";
  type: "market" | "limit";
}

File: backend/app/schemas/portfolio.py
from __future__ import annotations

from typing import Literal, Optional
from pydantic import BaseModel, Field


# ---------------------------------------------------------------------------
# Request
# ---------------------------------------------------------------------------

class TradeRequest(BaseModel):
    symbol: str = Field(..., min_length=1, max_length=10)
    qty: int = Field(..., gt=0)
    side: Literal["buy", "sell"]
    type: Literal["market", "limit"] = "market"
    limit_price: Optional[float] = None


class AlpacaConfigRequest(BaseModel):
    api_key: str = Field(..., min_length=1)
    api_secret: str = Field(..., min_length=1)
    paper_trading: bool = True


# ---------------------------------------------------------------------------
# Response
# ---------------------------------------------------------------------------

class PositionResponse(BaseModel):
    symbol: str
    qty: float
    avg_entry_price: float
    current_price: float
    market_value: float
    unrealized_pl: float
    unrealized_plpc: float
    side: str


class AccountResponse(BaseModel):
    equity: float
    buying_power: float
    day_pnl: float
    portfolio_value: float
    status: str


class TradeResponse(BaseModel):
    order_id: str
    symbol: str
    side: str
    qty: str
    type: str
    status: str
    filled_avg_price: Optional[str] = None
    submitted_at: str = ""


class AlpacaConfigResponse(BaseModel):
    is_connected: bool
    paper_trading: bool
    last_sync: Optional[str] = None
